bin/pycbc_make_skymap

```diff
@@ -115,137 +125,142 @@ def main(trig_time, mass1, mass2, spin1z, spin2z, f_low, f_upper, sample_rate,
         segment_length = 128
     logging.info('Using segment length: %s', segment_length)
 
-    gps_end_time = int(rough_time + pad // 2)
+    gps_end_time = int(mean_trig_time + pad / 2)
     gps_start_time = gps_end_time - segment_length
     logging.info("Using data: %s-%s", gps_start_time, gps_end_time)
 
     # if requested, remove detectors with missing/vetoed data
     # over the required segment
-    required_duration = gps_end_time - gps_start_time
+    required_duration = (gps_end_time + pad_data) - (gps_start_time - pad_data)
+    unavailable_ifos = set()
     for ifo in ifos:
         if detector_state is None or ifo not in detector_state:
             continue
         on_segs = dq.query_str(ifo, detector_state[ifo],
-                               gps_start_time, gps_end_time,
+                               gps_start_time - pad_data,
+                               gps_end_time + pad_data,
                                veto_definer=veto_definer,
+                               source=segment_source,
                                server=segment_server)
         if abs(on_segs) < required_duration:
             logging.info('Excluding %s due to missing or vetoed data', ifo)
-            ifos.remove(ifo)
+            unavailable_ifos.add(ifo)
+    ifos = sorted(set(ifos) - unavailable_ifos)
     if not ifos:
         raise RuntimeError('All detectors have been excluded due to '
                            'missing or vetoed data')
 
     highpass_frequency = int(f_low * 0.7)
     logging.info("Setting highpass: %s Hz", highpass_frequency)
 
-    for i, ifo in enumerate(ifos):
-        followup_data[ifo] = {}
+    procs = []
+    st_psd_paths = {}
+    st_out_paths = {}
+    for ifo in ifos:
+        # compose the command line for the single-template process
+        st_psd_paths[ifo] = os.path.join(
+                tmpdir, 'PSD_{}_{}.txt'.format(file_name_tag, ifo))
+        st_out_paths[ifo] = os.path.join(
+                tmpdir, 'SNRTS_{}_{}.hdf'.format(file_name_tag, ifo))
+
         command = ["pycbc_single_template",
```

Comment:
I wonder if we should have `make_skymap` take an ini file as input to take all these options (and more gracefully handle some of the other things like segment names, injection files, data choices etc.).

Then we could use some of the workflow stuff to setup the job and then run it immediately.

More than happy for you to push back and say "Do that in a separate PR!"

Reply:
Do that in a separate PR! :) But point taken, and I am happy to play with this for O4.